src/frontend/hooks/server/media.ts

```diff
@@ -1,52 +1,99 @@
 import {Observation} from '@mapeo/schema';
 import {BlobVariant} from '@mapeo/core/dist/types';
-import {useMutation, useQuery} from '@tanstack/react-query';
+import {useMutation, useQueries} from '@tanstack/react-query';
+import {SetRequired} from 'type-fest';
+import {URL} from 'react-native-url-polyfill';
 
-import {Photo} from '../../contexts/PhotoPromiseContext/types';
+import {DraftPhoto} from '../../contexts/PhotoPromiseContext/types';
 import {useProject} from './projects';
 
+type SavablePhoto = SetRequired<
+  Pick<DraftPhoto, 'originalUri' | 'previewUri' | 'thumbnailUri'>,
+  'originalUri'
+>;
+
 export function useCreateAttachmentsMutation() {
   const project = useProject();
 
   return useMutation({
-    mutationFn: async (photos: Photo[]) => {
+    mutationFn: async (photos: SavablePhoto[]) => {
       if (!project) throw new Error('Project instance does not exist');
 
       return Promise.all(
         photos.map(p => {
-          // TODO: Fix
+          const {originalUri, previewUri, thumbnailUri} = p;
+
           return project.$blobs.create(
-            {original: 'abc'},
-            {mimeType: 'img/png'},
+            {
+              original: new URL(originalUri).pathname,
+              preview: previewUri ? new URL(previewUri).pathname : undefined,
+              thumbnail: thumbnailUri
+                ? new URL(thumbnailUri).pathname
+                : undefined,
+            },
+            // TODO: DraftPhoto type should probably carry MIME type info that feeds this
+            // although backend currently only uses first part of path
+            {mimeType: 'image/jpeg'},
           );
         }),
       );
     },
   });
 }
 
-// TODO: Fix `type` issues
-export function useObservationAttachmentUrl(
-  attachment: Observation['attachments'][number],
-  // @ts-expect-error
-  variant: BlobVariant<(typeof attachment)['type']>,
+export function useAttachmentUrlQueries(
+  attachments: Observation['attachments'],
+  variant: BlobVariant<
+    Exclude<Observation['attachments'][number]['type'], 'UNRECOGNIZED'>
+  >,
 ) {
   const project = useProject();
 
-  return useQuery({
-    queryKey: ['attachmentUrl'],
-    queryFn: async () => {
-      if (!project) throw new Error('Project instance does not exist');
-      return project.$blobs.getUrl({
-        driveId: attachment.driveDiscoveryId,
-        // @ts-expect-error
-        type: attachment.type,
-        variant,
-      });
-    },
-    enabled: !!project,
-    // setting staleTime and CacheTime to infinity means it will only call the api once (aka photos do not get editted so we do not need to keep getting from the server)
-    staleTime: Infinity,
-    gcTime: Infinity,
+  return useQueries({
+    queries: attachments.map(attachment => {
+      return {
+        staleTime: Infinity,
+        gcTime: Infinity,
```

Comment:
is this correct? i kept this from the unused pre-existing implementation of this hook, but wasn't entirely sure if it's still relevant or correct to keep

Reply:
think i wrote it in the past with your guidance because i was unfamiliar with both options 😄 good to know that we probably don't need them